code/main.c

```diff
@@ -163,11 +167,28 @@ int main(int argc, char **argv)
     }
 
     finalize_galaxy_file(filenr);
-    free_tree_table();
+    free_tree_table(TreeType);
 
     printf("\ndone file %d\n\n", filenr);
   }
 
+  printf("Output 0 had %d galaxies\n", TotGalaxies[0]);
+
+/*
+  if(HDF5Output){
```

Comment:
I am in two minds -- how hard will it be to finish off the hdf5 writer? If too hard, we should remove the traces from the existing codebase, (including `HDF5OUTPUT`..

Reply:
In the scheme of things I don't think it will be too much; probably another day of work similar to the reader part.  However introducing the writer would also mean that ``allresults.py`` should be updated/an option added to handle HDF5 files.  I think that's where things get murky and we'd need to think about justifying the time investment.